From 2afb39374410d803dd287f83de44fbf7a4c2edc8 Mon Sep 17 00:00:00 2001
From: Kyungdahm Yun <tomyun@gmail.com>
Date: Thu, 27 Aug 2015 03:24:26 -0700
Subject: [PATCH] emudummy.c: change GAME_NO_SOUND to MACHINE_*

---
 src/emu/drivers/emudummy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/emu/drivers/emudummy.c b/src/emu/drivers/emudummy.c
index 4010fde..948be2f 100644
--- a/src/emu/drivers/emudummy.c
+++ b/src/emu/drivers/emudummy.c
@@ -33,4 +33,4 @@ ROM_START( __dummy )
 ROM_END
 
 
-GAME( 1900, __dummy, 0, __dummy, 0, driver_device, 0, ROT0, "(none)", "Dummy", GAME_NO_SOUND )
+GAME( 1900, __dummy, 0, __dummy, 0, driver_device, 0, ROT0, "(none)", "Dummy", MACHINE_NO_SOUND )
-- 
2.4.3

