diff -up mame-0.142u1/makefile.verbosebuild mame-0.142u1/makefile
--- mame-0.142u1/makefile.verbosebuild	2011-04-19 21:29:01.000000000 +0200
+++ mame-0.142u1/makefile	2011-04-19 21:31:40.000000000 +0200
@@ -297,12 +297,12 @@ BUILD_EXE = $(EXE)
 endif
 
 # compiler, linker and utilities
-AR = @ar
-CC = @gcc
-LD = @g++
+AR = ar
+CC = gcc
+LD = g++
 MD = -mkdir$(EXE)
-RM = @rm -f
-OBJDUMP = @objdump
+RM = rm -f
+OBJDUMP = objdump
 
 
 
@@ -755,12 +755,12 @@ $(OBJ)/%.s: $(SRC)/%.c | $(OSPREBUILD)
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
+	$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
 
 $(OBJ)/%.fh: $(SRC)/%.png $(PNG2BDC_TARGET) $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(PNG2BDC) $< $(OBJ)/temp.bdc
-	@$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
+	$(PNG2BDC) $< $(OBJ)/temp.bdc
+	$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
 
 $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
