diff -up mame-0.158/makefile.verbosebuild mame-0.158/makefile
--- mame-0.158/makefile.verbosebuild	2015-01-28 18:46:50.098531078 +0100
+++ mame-0.158/makefile	2015-01-28 18:51:09.123828941 +0100
@@ -360,14 +360,14 @@ endif
 
 # compiler, linker and utilities
 ifneq ($(TARGETOS),emscripten)
-AR = @ar
-CC = @gcc
-LD = @g++
+AR = ar
+CC = gcc
+LD = g++
 endif
 MD = -mkdir$(BUILD_EXE)
-RM = @rm -f
-OBJDUMP = @objdump
-PYTHON = @python
+RM = rm -f
+OBJDUMP = objdump
+PYTHON = python
 
 
 #-------------------------------------------------
@@ -956,28 +956,28 @@ $(OBJ)/%.o: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.o: $(OBJ)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.pp: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -E $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.s: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -S $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(SRC)/build/file2str.py
@@ -993,12 +993,12 @@ $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(DRIVLISTSRC): $(SRC)/$(TARGET)/$(SUBTARGET).lst $(MAKELIST_TARGET)
 	@echo Building driver list $<...
-	@$(MAKELIST) $< >$@
+	$(MAKELIST) $< >$@
 
 $(OBJ)/%.a:
 	@echo Archiving $@...
