diff -up mame-0.147u4/makefile.verbosebuild mame-0.147u4/makefile
--- mame-0.147u4/makefile.verbosebuild	2012-12-17 21:53:58.176154755 +0100
+++ mame-0.147u4/makefile	2012-12-17 21:55:26.559977075 +0100
@@ -307,13 +307,13 @@ BUILD_EXE = $(EXE)
 endif
 
 # compiler, linker and utilities
-AR = @ar
-CC = @gcc
-LD = @g++
+AR = ar
+CC = gcc
+LD = g++
 MD = -mkdir$(EXE)
-RM = @rm -f
-OBJDUMP = @objdump
-PYTHON = @python
+RM = rm -f
+OBJDUMP = objdump
+PYTHON = python
 
 
 #-------------------------------------------------
@@ -786,12 +786,12 @@ $(OBJ)/%.s: $(SRC)/%.c | $(OSPREBUILD)
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
+	$(FILE2STR) $< $@ layout_$(basename $(notdir $<))
 
 $(OBJ)/%.fh: $(SRC)/%.png $(PNG2BDC_TARGET) $(FILE2STR_TARGET)
 	@echo Converting $<...
-	@$(PNG2BDC) $< $(OBJ)/temp.bdc
-	@$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
+	$(PNG2BDC) $< $(OBJ)/temp.bdc
+	$(FILE2STR) $(OBJ)/temp.bdc $@ font_$(basename $(notdir $<)) UINT8
 
 $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
