diff -up mame-0.162/makefile.optflags mame-0.162/makefile
--- mame-0.162/makefile.optflags	2015-05-28 23:21:35.868882781 +0200
+++ mame-0.162/makefile	2015-05-28 23:23:02.109830454 +0200
@@ -51,6 +51,7 @@
 # MAP = 1
 # PROFILE = 1
 # ARCHOPTS =
+# OPT_FLAGS = 
 # LDOPTS =
 
 # USE_SYSTEM_LIB_EXPAT = 1
@@ -399,6 +400,10 @@ ifdef ARCHOPTS
 PARAMS += --ARCHOPTS='$(ARCHOPTS)'
 endif
 
+ifdef OPT_FLAGS
+PARAMS += --OPT_FLAGS='$(OPT_FLAGS)'
+endif
+
 ifdef MAP
 PARAMS += --MAP='$(MAP)'
 endif
diff -up mame-0.162/scripts/genie.lua.optflags mame-0.162/scripts/genie.lua
--- mame-0.162/scripts/genie.lua.optflags	2015-05-28 23:21:51.493054469 +0200
+++ mame-0.162/scripts/genie.lua	2015-05-28 23:24:07.232546065 +0200
@@ -164,6 +164,11 @@ newoption {
 }
 
 newoption {
+	trigger = "OPT_FLAGS",
+	description = "OPT_FLAGS.",
+}
+
+newoption {
 	trigger = "LDOPTS",
 	description = "Additional linker options",
 }
@@ -719,6 +724,11 @@ if _OPTIONS["OPTIMIZE"] then
 			_OPTIONS["ARCHOPTS"]
 		}
 	end
+	if _OPTIONS["OPT_FLAGS"] then
+		buildoptions {
+			_OPTIONS["OPT_FLAGS"]
+		}
+	end
 	if _OPTIONS["LTO"]=="1" then
 		buildoptions {
 			"-flto",
