diff -up mame-0.159/makefile.verbosebuild mame-0.159/makefile
--- mame-0.159/makefile.verbosebuild	2015-02-26 07:20:06.765476481 +0100
+++ mame-0.159/makefile	2015-02-26 07:27:46.388717313 +0100
@@ -363,14 +363,14 @@ endif
 
 # compiler, linker and utilities
 ifneq ($(TARGETOS),emscripten)
-AR = @ar
-CC = @gcc
-LD = @g++
+AR = ar
+CC = gcc
+LD = g++
 endif
 MD = -mkdir$(BUILD_EXE)
-RM = @rm -f
-OBJDUMP = @objdump
-PYTHON = @python
+RM = rm -f
+OBJDUMP = objdump
+PYTHON = python
 
 #-------------------------------------------------
 # form the name of the executable
@@ -975,28 +975,28 @@ $(OBJ)/%.o: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.o: $(OBJ)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.pp: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -E $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.s: $(SRC)/%.c | $(OSPREBUILD)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -S $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(OBJ)/%.lh: $(SRC)/%.lay $(SRC)/build/file2str.py
@@ -1012,7 +1012,7 @@ $(DRIVLISTOBJ): $(DRIVLISTSRC)
 	@echo Compiling $<...
 	$(CC) $(CDEFS) $(CFLAGS) -c $< -o $@
 ifdef CPPCHECK
-	@$(CPPCHECK) $(CPPCHECKFLAGS) $<
+	$(CPPCHECK) $(CPPCHECKFLAGS) $<
 endif
 
 $(DRIVLISTSRC): $(SRC)/$(TARGET)/$(SUBTARGET).lst $(SRC)/build/makelist.py
