diff -up mame-0.143u5/src/emu/cpu/m68000/m68kmake.c.stacksmash mame-0.143u5/src/emu/cpu/m68000/m68kmake.c
--- mame-0.143u5/src/emu/cpu/m68000/m68kmake.c.stacksmash	2011-09-08 07:20:36.000000000 +0200
+++ mame-0.143u5/src/emu/cpu/m68000/m68kmake.c	2011-09-08 07:22:01.909095642 +0200
@@ -81,7 +81,7 @@ static const char g_version[] = "4.80";
 #define MAX_BODY_LENGTH                 300	/* Number of lines in 1 function */
 #define MAX_REPLACE_LENGTH               30	/* Max number of replace strings */
 #define MAX_INSERT_LENGTH              5000	/* Max size of insert piece */
-#define MAX_NAME_LENGTH                  40	/* Max length of ophandler name */
+#define MAX_NAME_LENGTH                  96	/* Max length of ophandler name */
 #define MAX_SPEC_PROC_LENGTH              4	/* Max length of special processing str */
 #define MAX_SPEC_EA_LENGTH                5	/* Max length of specified EA str */
 #define EA_ALLOWED_LENGTH                11	/* Max length of ea allowed str */
