diff -up mame-0.171/makefile.systemlibs mame-0.171/makefile
--- mame-0.171/makefile.systemlibs	2016-02-25 20:05:32.207373656 +0100
+++ mame-0.171/makefile	2016-02-25 20:22:20.636524939 +0100
@@ -62,6 +62,7 @@
 # USE_SYSTEM_LIB_SQLITE3 = 1
 # USE_SYSTEM_LIB_PORTMIDI = 1
 # USE_SYSTEM_LIB_PORTAUDIO = 1
+# USE_SYSTEM_LIB_UV = 1
 
 # MESA_INSTALL_ROOT = /opt/mesa
 # SDL_INSTALL_ROOT = /opt/sdl2
@@ -390,6 +391,10 @@ ifndef USE_SYSTEM_LIB_PORTAUDIO
 PARAMS += --with-bundled-portaudio
 endif
 
+ifndef USE_SYSTEM_LIB_UV
+PARAMS += --with-bundled-libuv
+endif
+
 #-------------------------------------------------
 # distribution may change things
 #-------------------------------------------------
diff -up mame-0.171/scripts/genie.lua.systemlibs mame-0.171/scripts/genie.lua
--- mame-0.171/scripts/genie.lua.systemlibs	2016-02-25 20:05:32.017371824 +0100
+++ mame-0.171/scripts/genie.lua	2016-02-25 20:05:56.800610787 +0100
@@ -160,6 +160,11 @@ newoption {
 }
 
 newoption {
+    trigger = 'with-bundled-libuv',
+    description = 'Build bundled libuv library',
+}
+
+newoption {
 	trigger = "distro",
 	description = "Choose distribution",
 	allowed = {
diff -up mame-0.171/scripts/src/3rdparty.lua.systemlibs mame-0.171/scripts/src/3rdparty.lua
--- mame-0.171/scripts/src/3rdparty.lua.systemlibs	2016-02-25 20:05:32.071372345 +0100
+++ mame-0.171/scripts/src/3rdparty.lua	2016-02-25 20:05:56.801610797 +0100
@@ -989,6 +989,7 @@ end
 --------------------------------------------------
 -- libuv library objects
 --------------------------------------------------
+if _OPTIONS["with-bundled-libuv"] then
 project "uv"
 	uuid "cd2afe7f-139d-49c3-9000-fc9119f3cea0"
 	kind "StaticLib"
@@ -1160,6 +1161,11 @@ project "uv"
 		}
 	end
 
+else
+links {
+	"libuv",
+}
+end
 --------------------------------------------------
 -- HTTP parser library objects
 --------------------------------------------------
